src/System.CommandLine/Invocation/InvocationPipeline.cs

```diff
@@ -18,9 +18,28 @@ public InvocationPipeline(ParseResult parseResult)
 
         public async Task<int> InvokeAsync(IConsole console = null)
         {
-            var context = new InvocationContext(parseResult,
-                                                console);
+            var context = new InvocationContext(parseResult, console);
 
+            InvocationMiddleware invocationChain = BuildInvocationChain(context);
+
+            await invocationChain(context, invocationContext => Task.CompletedTask);
+
+            return GetResultCode(context);
+        }
+
+        public int Invoke(IConsole console = null)
+        {
+            var context = new InvocationContext(parseResult, console);
+
+            InvocationMiddleware invocationChain = BuildInvocationChain(context);
+
+            Task.Run(() => invocationChain(context, invocationContext => Task.CompletedTask)).Wait();
```

Comment:
Do **NOT** use `Wait()`! If anything this should be

`invocationChain(context, invocationContext => Task.CompletedTask).GetAwaiter().GetResult();`

That's the most correct and least error-prone way to execute asyncs synchronously.

Reply:
ref.: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#avoid-using-taskresult-and-taskwait

Reply:
I realize we are not in ASP.NET, so the issues David points out in regards to thread blocking can be ignored in this case.

However, `Task.Wait()` will throw an `AggregateException` if the invocation throws. `.GetAwaiter().GetResult()` will actually throw the original exception. 

This is also the reason why the compiler lowers `static async Task Main() {}` to `static void Main() => MainAsync().GetAwaiter().GetResult()`. If the async main throws, you still get the original exception thrown into your face, not a wrapping `AggregateException`.

Reply:
Fixed. Thank you.